tests/unit-tests/dds/DCPS/XTypes/DynamicDataImpl.cpp

```diff
@@ -745,6 +996,14 @@ void verify_string_union(DDS::DynamicType_var dt, const DataView& expected_cdr)
   EXPECT_EQ(ret, DDS::RETCODE_OK);
   ret = data.set_string_value(16, "def");
   EXPECT_EQ(ret, DDS::RETCODE_OK);
+
+  CORBA::Long disc_val;
+  EXPECT_EQ(DDS::RETCODE_OK, data.get_int32_value(disc_val, XTypes::DISCRIMINATOR_ID));
+  EXPECT_EQ(E_STRING8, disc_val);
+  char* str_val;
```

Comment:
Use a string manager?

Reply:
I'm not very familiar with string manager. Is that TAO's string manager?

Reply:
Yes, it's what the actual type of strings are in the TAO IDL mapping. I think they have implicit conversion operators for `char*&` and `const char*`for use in operation arguments, so that's what DynamicData uses.